### PR TITLE
HTMLProofer Retries & Timeout

### PR DESCRIPTION
**Pull request by PhilinTv**
## PR Description
- HTMLProofer was regularly failing on checking 3rd party links for actually available and healthy links, introducing flickery tests to the CI process
- Here we add 3x retry attempts and extend timeout from 30s to 60s

## Steps before you submit a PR
- Please add tests for the code you add if it's possible.
- Please check out our contribution guide: https://docs.spryker.com/docs/dg/dev/code-contribution-guide.html
- Add a `contribution-license-agreement.txt` file with the following content:
`I hereby agree to Spryker\'s Contribution License Agreement in https://github.com/spryker/spryker-docs/blob/HASH_OF_COMMIT_YOU_ARE_BASING_YOUR_BRANCH_FROM_MASTER_BRANCH/CONTRIBUTING.md.`

This is a mandatory step to make sure you are aware of the license agreement and agree to it. `HASH_OF_COMMIT_YOU_ARE_BASING_YOUR_BRANCH_FROM_MASTER_BRANCH` is a hash of the commit you are basing your branch from the master branch. You can take it from commits list of master branch before you submit a PR.

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

**Review by demkos — DISMISSED** [review]

**Review by demkos — DISMISSED** [review]

**Comment by gerner-spryker** [comment]
Also, please remove the "Steps before you submit a PR" section, as it is not applicable on you.

File: Rakefile [hunk 1]
```diff
@@ -12,6 +12,30 @@ end
 
 require 'html-proofer'
 
+# Method to run HTMLProofer with retries
+def run_htmlproofer_with_retry(directory, options, max_retries = 3, delay = 5)
+  options[:typhoeus] ||= {}
+  options[:typhoeus][:timeout] = 60
```
[the remark this answers is not in the document]
Comment (PhilinTv) [reply Rakefile:18]: 30 per default has a high chance of failure today. Raised 2x/.

File: Rakefile [hunk 2]
```diff
@@ -12,6 +12,30 @@ end
...
+  options[:typhoeus][:headers] = {
+    "User-Agent" => "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36"
```
[the remark this answers is not in the document]
Comment (PhilinTv) [reply Rakefile:20]: Just any, that is recongnized as a human. Most of wbesites passed the check now.

File: Rakefile [hunk 3]
```diff
@@ -72,7 +96,10 @@ commonOptions = {
     /code.visualstudio.com\/[\.\w\-\/\?]+/,
     /www.jetbrains.com\/[\.\w\-\/\?]+/,
     /docs.spring.io\/[\.\w\-\/\?]+/,
-    'http://redisdesktop.com/',
+    "http://redisdesktop.com/",
+    "https://developer.computop.com/display/EN/Test+Cards",
+    "https://www.centralbank.cy/",
```
[the remark this answers is not in the document]
Comment (PhilinTv) [reply Rakefile:101]: It appears we have an integration with them.

Their website works "sometimes a day" :D

**Review by gerner-spryker — APPROVED** [review]

**Pull request merged**